threat_model/README.md

```diff
@@ -0,0 +1,177 @@
+
+JSONDiff is accepts pull requests from the general public with a single committer. This model doesn't allow separation of duties. A single person owns the whole process of code creation, review, testing, and deployment.
+
+This threat is mitigated by having the entire project as open source. All code and processes may be reviewed by all third parties.
+
+### Third-party threats
+
+JSONDiff has only three external dependencies and doesn't use any code scanning for those threats.
+
+#### Threats
+1. JSONDiff doesn't provide an [SBOM](https://en.wikipedia.org/wiki/Software_supply_chain) file. This risk is mitigated by the very small number of dependencies used by JSONDiff.
+2. Google AdSense and Google Analytics could introduce malicious code. There's no good mitigation for this threat. JSONDiff just trusts those two Google projects. Google has their own process for threat modeling for those projects, but they must be loaded live and communicate back to Google for proper functionality.
+
+### Client-side threats
+
+JSONDiff is a client-side application and must consider client threats. Most client-side threats are not applicable given then very limited functionality of JSONDiff.
+
+#### CSRF and CSS attacks
+
+JSONDiff does not have any forms or submit any data, but it still has some potential vulnerability to [CSRF and CSS attacks](https://owasp.org/www-community/attacks/csrf). The main vector for those threats comes from the `proxy.php` file which can load potentially malicious files. Those files can be loaded via an URL and someone could be tricked into accessing that URL.
+
+JSONDiff mitigates this threat by ensuring that no files loaded from the `proxy.php` mechanism are ever executed. These files are only ever parsed using a custom JSON parser. 
+
+This can be verified with the following URL:
+
+```
+https://jsondiff.com/?left=https://jsondiff.com/evil.js&right=https://jsondiff.com/evil.json
+```
+
+### Stability, outages, and BCDR
+
+JSONDiff uses functional tests to ensure stability. High availability is provided by the service provider Media Temple. 
+
+The business continuity and datacenter recovery plan for JSONDiff is very simple. If a major outage occurs the service can be restored on any HTTP server that supports static files.
+
+The [JDDUpptime](https://github.com/zgrossbart/jdd-upptime) project monitors JSONDiff.com for outages.
+
+### Deployment
+
+JSONDiff deploys through sFTP to a file server.
+
+#### Threats
+
+1. If a malicious actor had the credentials they could attempt to access the sFTP site and upload malicious code to JSONDiff.com. This threat is mitigated by IP address source verification. All sFTP connections are verified by IP source address. All sFTP connections from a new source address require manual approval before the connection is granted.
+2. The sFTP user uses separate credentials from the main account and has only limited access. The sFTP user can not approve any new source addresses.
+
+### PII
```

Comment:
should we use full term for this, Privacy identifiable information?

Reply:
Good idea.  I made the change.